src/main/java/uk/gov/hmcts/reform/civil/helpers/bundle/BundleRequestMapper.java

```diff
@@ -868,6 +871,116 @@ private BundlingRequestDocument buildBundlingRequestDoc(String docName, Document
             .build();
     }
 
+    private void filterDocumentsUnbundledFolder(CaseData caseData) {
+
+        //Disclosure
+        filterGenericDisclosure(caseData.getDocumentForDisclosure());
```

Comment:
Is filtering enough, or do we also need to save it?
For example, is this enough?:
`uploadEvidenceExpert.stream().filter(caseDocumentElement -> (nonNull(caseDocumentElement.getValue().getExpertDocument().getCategoryID())
                && !caseDocumentElement.getValue().getExpertDocument().getCategoryID().equals("UnbundledFolder")));`
Or does it need to be something like: 
`List<> filteredExpertEvidence = uploadEvidenceExpert.stream().filter(caseDocumentElement -> (nonNull(caseDocumentElement.getValue().getExpertDocument().getCategoryID())
                && !caseDocumentElement.getValue().getExpertDocument().getCategoryID().equals("UnbundledFolder"))).toList();
caseDataBuilder.documentExpertReport(filteredExpertEvidence)`

I'm not sure - that's why I'm asking.

Reply:
It should be work because we are using the same object across all journey, let me create a specific unit test for that.

Reply:
it works but due to sonar issue I have refactored and tests works fine also... please have a look